### PR TITLE
feat: adds next-tinacms-markdown package

### PR DESCRIPTION
**Pull request by kendallstrautman**
This PR adds the `next-tinacms-markdown` package with helpers for editing Markdown in Next.js 
- `useMarkdownForm`
- `useLocalMarkdownForm`
- `useGlobalMarkdownForm`

It also adds the `demo-next` package for testing out next+tina development features.

### CODE REVIEW

**Comment by kendallstrautman** [comment]
Still Todo: 
- Need to add a few helpers for global, inline and render props etc. 
- Figure out the bootstrap error

**Comment by kendallstrautman** [comment]
I don't understand where those generated style component files came from when I merged master. Perhaps something left over when I was messing around on the `tinacms-ui` branch?

File: packages/next-tinacms-markdown/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "next-tinacms-markdown",
+  "version": "0.1.0-alpha.0",
+  "license": "Apache-2.0",
+  "author": "Nolan Phillips",
```
Comment (ncphillips) [inline packages/next-tinacms-markdown/package.json]: No I'm not 😉

File: packages/next-tinacms-markdown/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "test": "tsdx test --env=jsdom",
+    "lint": "tsdx lint"
+  },
+  "peerDependencies": {
+    "next": ">=9",
+    "react": ">=16",
+    "tinacms": ">=0.11"
+  },
+  "dependencies": {
+    "next": ">=9",
```
Comment (ncphillips) [inline packages/next-tinacms-markdown/package.json]: `next`, `react`, and `tinacms` should just be in `peerDependencies` and `devDependencies`.

I assume this came from `next-tinacms-json`?

File: packages/next-tinacms-markdown/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# next-tinacms-markdown
+
+This package provides helpers for editing content stored in Markdown files in the local filesystem.
+
+- `useMarkdownForm`
+- `useLocalMarkdownForm`
```
Comment (ncphillips) [inline packages/next-tinacms-markdown/README.md]:
```suggestion
- `useLocalMarkdownForm`
- `useGlobalMarkdownForm`
```

File: packages/next-tinacms-markdown/src/use-markdown-form.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ */
+export function useMarkdownForm (
+  markdownFile: MarkdownFile,
+  options: Options = {}
+) {
+  const cms = useCMS()
+
+  const [valuesInGit, setValuesInGit] = useState()
+
+  const valuesOnDisk = useMemo(
```
Comment (ncphillips) [inline packages/next-tinacms-markdown/src/use-markdown-form.ts]: I vaguely think the `useMemo` is unnecessary here. That callback isn't doing any processing, just creating a new object. I think `const valuesOnDisk = markdownFile` would be fine. 

I like making the new variable just for the sake of making it easier to understand.

I could be wrong on this. We could do a quick check to see if it causes a bunch of unnecessary rendering without the `useMemo`

File: packages/next-tinacms-markdown/src/use-markdown-form.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  const writeToDisk = useCallback(formState => {
+    cms.api.git.writeToDisk({
+      fileRelativePath: markdownFile.fileRelativePath,
+      content: toMarkdownString(formState.values),
+    })
+  }, [])
+
+  useWatchFormValues(form, writeToDisk)
+
+  return [values || {markdownBody: markdownFile.markdownBody, frontmatter: markdownFile.frontmatter}, form]
```
Comment (ncphillips) [inline packages/next-tinacms-markdown/src/use-markdown-form.ts]:
```suggestion
  return [values || markdownFile, form]
```

**Comment by ncphillips** [comment]
Woops! I realized this was a draft PR _after_ I added my comments. 😬

File: packages/next-tinacms-markdown/package.json (hunk 2 above)
Comment (kendallstrautman) [reply packages/next-tinacms-markdown/package.json]: oh yeah I just copied it over

File: packages/next-tinacms-markdown/src/use-markdown-form.ts (hunk 4 above)
Comment (kendallstrautman) [reply packages/next-tinacms-markdown/src/use-markdown-form.ts]: I was having a hard time getting it to work without `useMemo` but I will do some more testing

**Comment by kendallstrautman** [comment]
haha all good!

File: packages/next-tinacms-markdown/package.json (hunk 2 above)
Comment (kendallstrautman) [reply packages/next-tinacms-markdown/package.json]: is `gray-matter` okay being in `dependencies`? I saw that's how it was configured for `gatsby-tinacms-remark`

Comment (ncphillips) [reply packages/next-tinacms-markdown/package.json]: Yup!

Here's a pseudocode rule of thumb:
```
if package_X is a plugin for package_Y
then package_Y is a peerDependency of package_X
```

File: packages/next-tinacms-markdown/src/use-markdown-form.ts (hunk 4 above)
Comment (ncphillips) [reply packages/next-tinacms-markdown/src/use-markdown-form.ts]: Okay cool! If that's the case then I would just add `markdownFile.fileRelativePath` to the dependency array

File: packages/next-tinacms-markdown/package.json (hunk 2 above)
Comment (kendallstrautman) [reply packages/next-tinacms-markdown/package.json]: The build fails now without `tinacms` in `dependencies`...which makes sense. Thoughts on how to fix?

Comment (ncphillips) [reply packages/next-tinacms-markdown/package.json]: If it's in `peerDependencies` ti should also be in `devDependencies`

**Comment by kendallstrautman** [comment]
Going to add inline helpers in another PR, wanted to get all these changes in 👌🏼

**Pull request merged**